[2024-11-04T00:28:19.860+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: DAG_Data_Preprocessing.feature_selection_task manual__2024-11-04T00:23:06.270555+00:00 [queued]>
[2024-11-04T00:28:19.882+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: DAG_Data_Preprocessing.feature_selection_task manual__2024-11-04T00:23:06.270555+00:00 [queued]>
[2024-11-04T00:28:19.883+0000] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2024-11-04T00:28:19.883+0000] {taskinstance.py:1280} INFO - Starting attempt 2 of 2
[2024-11-04T00:28:19.883+0000] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2024-11-04T00:28:19.902+0000] {taskinstance.py:1300} INFO - Executing <Task(PythonOperator): feature_selection_task> on 2024-11-04 00:23:06.270555+00:00
[2024-11-04T00:28:19.918+0000] {standard_task_runner.py:55} INFO - Started process 77 to run task
[2024-11-04T00:28:19.921+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'DAG_Data_Preprocessing', 'feature_selection_task', 'manual__2024-11-04T00:23:06.270555+00:00', '--job-id', '180', '--raw', '--subdir', 'DAGS_FOLDER/***dag.py', '--cfg-path', '/tmp/tmpildsfs32']
[2024-11-04T00:28:19.924+0000] {standard_task_runner.py:83} INFO - Job 180: Subtask feature_selection_task
[2024-11-04T00:28:20.008+0000] {task_command.py:388} INFO - Running <TaskInstance: DAG_Data_Preprocessing.feature_selection_task manual__2024-11-04T00:23:06.270555+00:00 [running]> on host b77d411e2e76
[2024-11-04T00:28:20.204+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=House_Price_Prediction Team
AIRFLOW_CTX_DAG_ID=DAG_Data_Preprocessing
AIRFLOW_CTX_TASK_ID=feature_selection_task
AIRFLOW_CTX_EXECUTION_DATE=2024-11-04T00:23:06.270555+00:00
AIRFLOW_CTX_TRY_NUMBER=2
AIRFLOW_CTX_DAG_RUN_ID=manual__2024-11-04T00:23:06.270555+00:00
[2024-11-04T00:28:21.305+0000] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 192, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/airflowdag.py", line 188, in feature_selection_callable
    selected_features = select_correlated_features(df_encoded, numerical_features, target, threshold=0.3)
  File "/opt/airflow/dags/src/feature_select.py", line 20, in select_correlated_features
    df = pd.read_json(encoded_data)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/util/_decorators.py", line 207, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 607, in read_json
    encoding_errors=encoding_errors,
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 675, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 718, in _get_data_from_filepath
    errors=self.encoding_errors,
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/common.py", line 585, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/common.py", line 962, in _is_binary_mode
    return isinstance(handle, binary_classes) or "b" in getattr(handle, "mode", mode)
TypeError: argument of type 'method' is not iterable
[2024-11-04T00:28:21.344+0000] {taskinstance.py:1323} INFO - Marking task as FAILED. dag_id=DAG_Data_Preprocessing, task_id=feature_selection_task, execution_date=20241104T002306, start_date=20241104T002819, end_date=20241104T002821
[2024-11-04T00:28:21.368+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 180 for task feature_selection_task (argument of type 'method' is not iterable; 77)
[2024-11-04T00:28:21.421+0000] {local_task_job.py:208} INFO - Task exited with return code 1
[2024-11-04T00:28:21.507+0000] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
